hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/amqp/DmfReceiverService.java

```diff
@@ -182,37 +185,89 @@ private void handleEventMessage(final Message message, final String thingId) {
         case REQUEST_ATTRIBUTES_UPDATE:
             handleAttributeUpdateRequest(message, thingId);
             break;
+        case MULTI_ACTION:
+            handleMultiActionRequest(message, thingId);
+            break;
         default:
             LOGGER.info("No valid event property.");
             break;
         }
     }
 
-    private void handleAttributeUpdateRequest(final Message message, final String thingId) {
-        final MessageProperties messageProperties = message.getMessageProperties();
-        final Map<String, Object> headers = messageProperties.getHeaders();
-        final String tenant = (String) headers.get(MessageHeaderKey.TENANT);
+    private void handleMultiActionRequest(final Message message, final String thingId) {
+
+        final DmfMultiActionRequest multiActionRequest = convertMessage(message, DmfMultiActionRequest.class);
+        final String tenant = getTenant(message);
+
+        final DmfMultiActionRequest.DmfMultiActionElement actionElement =
+                multiActionRequest.getElements().get(0);
+
+        processAction(thingId, tenant, actionElement);
+    }
+
+    private void processAction(final String thingId, final String tenant,
+            final DmfMultiActionRequest.DmfMultiActionElement actionElement) {
+        final EventTopic eventTopic = actionElement.getTopic();
+        final DmfActionRequest action = actionElement.getAction();
+        final long actionId = action.getActionId();
+
+        if(openActions.stream().anyMatch(id -> id == actionId)) {
+            return;
+        }
+
+        openActions.add(actionId);
 
+        switch (eventTopic) {
+            case DOWNLOAD :
+            case DOWNLOAD_AND_INSTALL :
+                if (action instanceof DmfDownloadAndUpdateRequest) {
+                    processUpdate(thingId, eventTopic, tenant, (DmfDownloadAndUpdateRequest) action);
+                }
+                break;
+            case CANCEL_DOWNLOAD :
+                processCancelDownloadAction(thingId, tenant, action.getActionId());
+                break;
+            default :
+                LOGGER.info("No valid event property in MULTI_ACTION.");
```

Comment:
in this case, the action is added to the openActions but not removed. Maybe we can use the `logAndThrowMessageError` to send an error to the server, that way we don't have to keep track of this kind of actions